libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFPasscodeViewController.m

```diff
@@ -347,201 +387,325 @@ -(BOOL) shouldAutorotate {
+    [self.passcodeInstructionsLabel setFont:self.viewConfig.instructionFont];
+    [self.passcodeInstructionsLabel setHidden:NO];
+    [self.passcodeTextView setHidden:NO];
+   
+    [self.passcodeTextView refreshView];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.passcodeTextView becomeFirstResponder];
+    });
 }
 
-- (void)addPasscodeCreationNav
+- (void)showVerifyPasscode
 {
-    UIBarButtonItem *bbi = [[UIBarButtonItem alloc] initWithTitle:[SFSDKResourceUtils localizedString:@"nextScreenNavButtonTitle"]
-                                                            style:UIBarButtonItemStylePlain
-                                                           target:self
-                                                           action:@selector(finishedInitialPasscode)];
-    [self.navigationItem setRightBarButtonItem:bbi];
-    [self.navigationItem setLeftBarButtonItem:nil];
-    [self.navigationItem setTitle:[SFSDKResourceUtils localizedString:@"createPasscodeNavTitle"]];
+    [self hideAll];
+    [self.navigationItem setTitle:[SFSDKResourceUtils localizedString:@"verifyPasscodeNavTitle"]];
+    [self.passcodeInstructionsLabel setText:[SFSDKResourceUtils localizedString:@"passcodeVerifyInstructions"]];
+    [self.passcodeInstructionsLabel setFont:self.viewConfig.instructionFont];
+    [self.passcodeInstructionsLabel setHidden:NO];
+    [self.passcodeTextView setHidden:NO];
+    
+    if (self.passcodeLength == 0) {
+        [self.verifyPasscodeButton setHidden:NO];
+    }
+    [self.passcodeTextView refreshView];
+    dispatch_async(dispatch_get_main_queue(), ^{
+        [self.passcodeTextView becomeFirstResponder];
+    });
 }
 
-- (void)addPasscodeConfirmNav
+- (void)verifyPasscode
 {
-    UIBarButtonItem *bbi = [[UIBarButtonItem alloc] initWithBarButtonSystemItem:UIBarButtonSystemItemDone
-                                                                         target:self
-                                                                         action:@selector(finishedConfirmPasscode)];
-    [self.navigationItem setRightBarButtonItem:bbi];
-    bbi = [[UIBarButtonItem alloc] initWithTitle:[SFSDKResourceUtils localizedString:@"prevScreenNavButtonTitle"]
-                                           style:UIBarButtonItemStylePlain
-                                          target:self
-                                          action:@selector(resetInitialCreateView)];
-    [self.navigationItem setLeftBarButtonItem:bbi];
-    [self.navigationItem setTitle:[SFSDKResourceUtils localizedString:@"confirmPasscodeNavTitle"]];
-    [self.passcodeField becomeFirstResponder];
+    if ([[SFPasscodeManager sharedManager] verifyPasscode:self.passcodeTextView.passcodeInput]) {
+        if ([self.passcodeTextView isFirstResponder])
+            [self.passcodeTextView resignFirstResponder];
+        if ([self canShowBiometricEnrollmentScreen]) {
```

Comment:
Shouldn't biometric setup happen once they have successfully verified the passcode?